prow.go

```diff
@@ -381,18 +366,51 @@ func (m *jobManager) newJob(job *Job) error {
 				if err != nil {
 					return fmt.Errorf("unable to reformat child job for %#v: %v", ref, err)
 				}
-				prow.SetJobEnvVar(&pj.Spec, fmt.Sprintf("CONFIG_SPEC_%d", index), string(data))
+				configSpecs = append(configSpecs, string(data))
 
 				data, err = json.MarshalIndent(JobSpec{Refs: &ref}, "", "  ")
 				if err != nil {
 					return fmt.Errorf("unable to reformat child job for %#v: %v", ref, err)
 				}
-				prow.SetJobEnvVar(&pj.Spec, fmt.Sprintf("JOB_SPEC_%d", index), string(data))
+				jobSpecs = append(jobSpecs, string(data))
 
 				index++
 			}
 		}
 
+		switch container := &pj.Spec.PodSpec.Containers[0]; {
+		case reflect.DeepEqual(container.Command, []string{"ci-operator"}):
+			var args []string
+			for _, arg := range container.Args {
+				if strings.HasPrefix(arg, "--namespace") {
+					continue
+				}
+				args = append(args, arg)
+			}
+			args = append(args, fmt.Sprintf(`--namespace=$(NAMESPACE)`))
+
+			// Assemble script from scriptPre / scriptChild / scriptPost
+			script := scriptPre
+			for i := 0; i < len(configSpecs); i++ {
+				configSpec := configSpecs[i]
+				jobSpec := jobSpecs[i]
+				script += fmt.Sprintf(scriptChild, i, jobSpec, configEnvVar, configSpec)
```

Comment:
This doesn't read any easier and it's actually scaring me more.  We don't need to pass config_spec to children, you can always pass CONFIG_SPEC as UNRESOLVED_CONFIG and it works fine.  So just make the script always use UNRESOLVED_CONFIG and convert CONFIG_SPEC to UNRESOLVED_CONFIG in the job definition.

That removes the need to change generation of the script file (you can rename the env var up front if the job definition has CONFIG_SPEC)

Reply:
Oh. so UNRESOLVED_CONFIG would be accepted when no CONFIG_SPEC present? Okay, that makes it much easier